kaldb/src/test/java/com/slack/kaldb/clusterManager/ReplicaEvictionServiceTest.java

```diff
@@ -0,0 +1,814 @@
+            .setEventAggregationSecs(10)
+            .setScheduleInitialDelayMins(0)
+            .build();
+
+    new ReplicaEvictionService(
+            cacheSlotMetadataStore, replicaMetadataStore, managerConfig, meterRegistry)
+        .scheduler();
+  }
+
+  @Test
+  public void shouldHandleNoReplicasOrAssignedSlots() {
+    KaldbConfigs.ManagerConfig.ReplicaEvictionServiceConfig replicaEvictionServiceConfig =
+        KaldbConfigs.ManagerConfig.ReplicaEvictionServiceConfig.newBuilder()
+            .setSchedulePeriodMins(10)
+            .build();
+
+    KaldbConfigs.ManagerConfig managerConfig =
+        KaldbConfigs.ManagerConfig.newBuilder()
+            .setReplicaEvictionServiceConfig(replicaEvictionServiceConfig)
+            .setEventAggregationSecs(10)
+            .setScheduleInitialDelayMins(10)
+            .build();
+
+    ReplicaEvictionService replicaEvictionService =
+        new ReplicaEvictionService(
+            cacheSlotMetadataStore, replicaMetadataStore, managerConfig, meterRegistry);
+
+    int replicasMarked = replicaEvictionService.markReplicasForEviction();
+    assertThat(replicasMarked).isEqualTo(0);
+
+    assertThat(
+            MetricsUtil.getCount(
+                ReplicaEvictionService.REPLICA_MARK_EVICT_SUCCEEDED, meterRegistry))
+        .isEqualTo(0);
+    assertThat(
+            MetricsUtil.getCount(ReplicaEvictionService.REPLICA_MARK_EVICT_FAILED, meterRegistry))
+        .isEqualTo(0);
+    assertThat(
+            MetricsUtil.getTimerCount(
+                ReplicaEvictionService.REPLICA_MARK_EVICT_TIMER, meterRegistry))
+        .isEqualTo(1);
+  }
+
+  @Test
+  public void shouldHandleNoExpiredCacheSlots() {
+    KaldbConfigs.ManagerConfig.ReplicaEvictionServiceConfig replicaEvictionServiceConfig =
+        KaldbConfigs.ManagerConfig.ReplicaEvictionServiceConfig.newBuilder()
+            .setSchedulePeriodMins(10)
+            .build();
+
+    KaldbConfigs.ManagerConfig managerConfig =
+        KaldbConfigs.ManagerConfig.newBuilder()
+            .setReplicaEvictionServiceConfig(replicaEvictionServiceConfig)
+            .setEventAggregationSecs(10)
+            .setScheduleInitialDelayMins(10)
+            .build();
+
+    ReplicaEvictionService replicaEvictionService =
+        new ReplicaEvictionService(
+            cacheSlotMetadataStore, replicaMetadataStore, managerConfig, meterRegistry);
+
+    List<ReplicaMetadata> replicas = new ArrayList<>();
+    for (int i = 0; i < 5; i++) {
+      ReplicaMetadata replicaMetadata =
+          new ReplicaMetadata(
+              UUID.randomUUID().toString(),
+              UUID.randomUUID().toString(),
+              Instant.now().toEpochMilli(),
+              Instant.now().plusSeconds(60).toEpochMilli());
+      replicas.add(replicaMetadata);
+      replicaMetadataStore.create(replicaMetadata);
+    }
+
+    List<CacheSlotMetadata> cacheSlots = new ArrayList<>();
+    for (int i = 0; i < 3; i++) {
+      CacheSlotMetadata cacheSlotMetadata =
```

Comment:
It may be better to add a unit test with mixed cache slot states and make sure those replicas are unaffected. Alternatively, it may be simpler to refactor the filter function into a predicate and add a unit test for that.

Reply:
Done.